### PR TITLE
fix: give breadcrumb item in overflow click proxy

### LINKED ISSUE #8521: Breadcrumb overflow items do not navigate on click
#### Is this a bug, enhancement, or feature request?
bug

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)
0.35.0+

#### If this is a bug, please provide steps for reproducing it.
1. Open https://sap.github.io/fundamental-ngx/#/core/breadcrumb
2. Decrease screen width until an overflow menu gets shown
3. Open overflow menu 
4. Breadcrumbs that are displayed in the overflow do not navigate on click

### Screenshot
breadcrumb-click

### PR DESCRIPTION
**Pull request by NancyKunath**
## Related Issue(s)
closes #8521

## Description
Breadcrumb items in overflow are now clickable (again).

##### During Implementation

1. Visual Testing:

-   [x] visual misalignments/updates
-   [x] check Light/Dark/HCB/HCW themes
-   [x] RTL/LTR - proper rendering and labeling
-   [x] responsiveness(resize)
-   [x] Content Density (Cozy/Compact/(Condensed))
-   [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [x] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [na] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [na] Run npm run build-pack-library and test in external application
-   [na] update `README.md`
-   [na] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

**Comment by g-cheishvili** [comment]
I checked https://sap.github.io/fundamental-ngx/#/core/bar and it works as expected(navigates without issue)

**Comment by NancyKunath** [comment]
I have added a gif showing the bug to the issue [#8521](https://github.com/SAP/fundamental-ngx/issues/8521). Please have a look.

**Comment by g-cheishvili** [comment]
I checked it out. You are right.

Just solution is not quite on point. `_needsClickProxy` is an internal thing, its usage should be changed, instead of name, underscore is important:

this should be in libs/core/src/lib/breadcrumb/breadcrumb.component.ts:125
```ts
itemClicked(breadcrumbItem: BreadcrumbItemComponent, $event: any): void {
      if (breadcrumbItem._needsClickProxy) {
          $event.preventDefault();
          breadcrumbItem.breadcrumbLink.elementRef().nativeElement.click();
      }
  }
```

**Comment by NancyKunath** [comment]
Okay. I added the underscore again.

**Review by N1XUS — APPROVED** [review]

File: libs/core/src/lib/breadcrumb/breadcrumb.component.ts [hunk 1]
```diff
@@ -123,7 +123,7 @@ export class BreadcrumbComponent implements OnInit, AfterViewInit {
      * if original element had router link we are proxying click to that element
      * */
     itemClicked(breadcrumbItem: any, $event: any): void {
```
Comment (g-cheishvili) [inline libs/core/src/lib/breadcrumb/breadcrumb.component.ts:125]: Please, add the type too, so it does not happen again

**Review by g-cheishvili — CHANGES_REQUESTED** [review]

File: libs/core/src/lib/breadcrumb/breadcrumb.component.ts (hunk 1 above)
Comment (NancyKunath) [reply libs/core/src/lib/breadcrumb/breadcrumb.component.ts]: Done 👍

**Review by mikerodonnell89 — APPROVED** [review]

**Review by platon-rov — APPROVED** [review]

**Review by g-cheishvili — APPROVED** [review]

**Comment by mikerodonnell89** [comment]
Closing in favor of #8599

**Pull request closed without merging**